aws/retry/metadata.go

```diff
@@ -0,0 +1,37 @@
+package retry
+
+import "github.com/aws/smithy-go/middleware"
+
+type requestAttemptsKey struct {
+}
+
+type requestAttempts struct {
+
+	// Attempts is a slice consisting metadata from all request attempts.
+	// Attempts are stored in last in first order i.e. the last attempt
+	// would be at the top.
+	Attempts []requestAttempt
+}
+
+type requestAttempt struct {
```

Comment:
Should these two types be exported?

Reply:
If the intention is to let customers access these values programmatically, then yes.